crates/wasmtime-provider/examples/wasmtime-hash-mreplace.rs

```diff
@@ -0,0 +1,101 @@
+#![allow(unused_imports)]
+use console::{style, Term};
+use log::{debug, error, info, warn};
+use serde::{Deserialize, Serialize};
+use smallvec::{smallvec, SmallVec};
+use std::fs::File;
+use std::io::Write;
+use std::{io::Read, time::Instant};
+use wapc_codec::messagepack::{deserialize, serialize};
+
+//simple struct to pass to wasm module and calc hash inside
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+struct PersonSend {
+  first_name: String,
+}
+// recv struct
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+struct PersonHashedRecv {
+  first_name: String,
+  hash: u64,
+}
+
+use wapc::WapcHost;
+use wasmtime_provider::WasmtimeEngineProviderBuilder;
+
+pub fn main() -> Result<(), wapc::errors::Error> {
+  env_logger::init();
+  println!("{}", style("starting app!").yellow());
+  let name = &std::env::args().nth(1).expect("pass some name to serde");
+  let module_bytes1 = std::fs::read("../../wasm/crates/wasm-calc-hash/module1/build/module1_hash.wasm")
+    .expect("WASM module 1 could not be read, run example from wasmtime-provider folder"); // read module 1
+  let module_bytes2 = std::fs::read("../../wasm/crates/wasm-calc-hash/module2/build/module2_hash.wasm")
+    .expect("WASM module 2 could not be read, run example from wasmtime-provider folder"); // read module 2
+  let func = "serdes_example".to_string();
+  let engine = WasmtimeEngineProviderBuilder::new()
+    .module_bytes(&module_bytes1)
+    .build()?;
+  assert_ne!(module_bytes1, module_bytes2); // test modules binaries not equal
+  let host = WapcHost::new(Box::new(engine), Some(Box::new(host_callback)))?;
+  println!(
+    "{} {}",
+    style("Calling guest (wasm) function ").cyan(),
+    style(&func).cyan()
+  );
```

Comment:
nitpicking: also in this case, we could use a vanilla `println` to avoid the `console` dependency